elements/lisk-transactions/src/15_proof_of_misbehavior_transaction.ts

```diff
@@ -0,0 +1,442 @@
+			b2 = asset.header1;
+		}
+
+		if (
+			!(
+				b1.maxHeightPrevoted === b2.maxHeightPrevoted && b1.height >= b2.height
+			) &&
+			!(b1.height > b2.maxHeightPreviouslyForged) &&
+			!(b1.maxHeightPrevoted > b2.maxHeightPrevoted)
+		) {
+			errors.push(
+				new TransactionError(
+					'Blockheaders are not contradicting as per BFT violation rules.',
+					this.id,
+					'.asset.header1',
+				),
+			);
+		}
+
+		return errors;
+	}
+
+	protected async applyAsset(
+		store: StateStore,
+	): Promise<ReadonlyArray<TransactionError>> {
+		const errors = [];
+		const currentHeight = store.chain.lastBlockHeader.height + 1;
+		const delegateId = getAddressFromPublicKey(
+			this.asset.header1.generatorPublicKey,
+		);
+		const delegateAccount = await store.account.get(delegateId);
+		const senderAccount = await store.account.get(this.senderId);
+		const { networkIdentifier } = store.chain;
+		/*
+			|header1.height - h| < 260,000.
+			|header2.height - h| < 260,000.
+		*/
+
+		// tslint:disable-next-line no-magic-numbers
+		if (Math.abs(this.asset.header1.height - currentHeight) >= 260000) {
+			errors.push(
+				new TransactionError(
+					'Difference between header1.height and current height must be less than 260000.',
+					this.id,
+					'.asset.header1',
+					this.asset.header1.height,
+				),
+			);
+		}
+
+		// tslint:disable-next-line no-magic-numbers
+		if (Math.abs(this.asset.header2.height - currentHeight) >= 260000) {
+			errors.push(
+				new TransactionError(
+					'Difference between header2.height and current height must be less than 260000.',
+					this.id,
+					'.asset.header2',
+					this.asset.header2.height,
+				),
+			);
+		}
+
+		/*
+			Check if delegate is eligible to be punished
+		*/
+
+		if (!delegateAccount.isDelegate || !delegateAccount.username) {
+			errors.push(
+				new TransactionError(
+					'Account is not a delegate',
+					this.id,
+					'.asset.header1.generatorPublicKey',
+				),
+			);
+		}
+
+		if (delegateAccount.delegate.isBanned) {
+			errors.push(
+				new TransactionError(
+					'Cannot apply proof-of-misbehavior. Delegate is banned.',
+					this.id,
+					'.asset.header1.generatorPublicKey',
+					this.asset.header1.generatorPublicKey,
+				),
+			);
+		}
+
+		if (
+			isPunished(
+				delegateAccount,
+				delegateAccount,
+				store.chain.lastBlockHeader.height,
+			)
+		) {
+			errors.push(
+				new TransactionError(
+					'Cannot apply proof-of-misbehavior. Delegate is already punished. ',
+					this.id,
+					'.asset.header1.generatorPublicKey',
+					this.asset.header1.generatorPublicKey,
+				),
+			);
+		}
+
+		/* 
+			Check block signatures validity 
+		*/
+
+		const blockHeader1Bytes = Buffer.concat([
+			Buffer.from(networkIdentifier, 'hex'),
+			getBlockBytes(this.asset.header1),
+		]);
+		const blockHeader2Bytes = Buffer.concat([
+			Buffer.from(networkIdentifier, 'hex'),
+			getBlockBytes(this.asset.header2),
+		]);
+
+		const { valid: validHeader1Signature } = validateSignature(
```

Comment:
this might be misleading because the 3rd argument name is `transactionBytes`. 
Maybe create a new function or Update the argument name to `bytes`?